crates/shared/src/persistence/sqlite.rs

```diff
@@ -0,0 +1,211 @@
+use super::BuilderPersistence;
+use anyhow::Result;
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use sqlx::Row;
+use sqlx::SqlitePool;
+use std::time::{Instant, SystemTime};
+
+#[allow(unused_imports)]
+use super::get_sqlite_test_db_path;
+
+#[derive(Debug)]
+pub struct SqliteTxnDb {
+    pool: SqlitePool,
+}
+impl SqliteTxnDb {
+    #[allow(dead_code)]
+    async fn new(database_url: String) -> Result<Self, sqlx::Error> {
+        let pool = SqlitePool::connect(&database_url).await?;
+        // it will handle the default CURRENT_TIMESTAMP automatically and assign to transaction's created_at
+        sqlx::query(
+            r#"
+            CREATE TABLE IF NOT EXISTS transactions (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                tx_data BLOB NOT NULL,
+                created_at DATETIME DEFAULT CURRENT_TIMESTAMP
+            );
+            "#,
+        )
+        .execute(&pool)
+        .await?;
+        Ok(Self { pool })
+    }
+
+    #[allow(dead_code)]
+    async fn clear(&self) -> Result<(), sqlx::Error> {
+        // Execute a SQL statement to delete all rows from the `transactions` table
+        sqlx::query("DELETE FROM transactions")
+            .execute(&self.pool)
+            .await?;
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl BuilderPersistence for SqliteTxnDb {
+    async fn append(&self, tx_data: Vec<u8>) -> Result<(), sqlx::Error> {
+        sqlx::query(
+            r#"
+            INSERT INTO transactions (tx_data) VALUES (?);
+            "#,
+        )
+        .bind(tx_data)
+        .execute(&self.pool)
+        .await?;
+        Ok(())
+    }
+
+    async fn load(&self, timeout_after: Instant) -> Result<Vec<Vec<u8>>, sqlx::Error> {
+        // Convert Instant to SystemTime
+        let now = SystemTime::now();
+        let elapsed = timeout_after.elapsed();
+        let target_time = now - elapsed;
```

Comment:
This might be the most straightforward way as far as I know. The reason for this is that Instant represents a point in time relative to an unspecified reference (it doesn't have a fixed "epoch" like SystemTime), while SystemTime represents an absolute point in time from the UNIX epoch. So without using `now - elapsed`, we want to first obtaining the system time at the moment of the Instant's creation